x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorer.java

```diff
@@ -56,22 +56,12 @@ public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext r
             throw new IllegalStateException("local model reference is null, missing rewriteAndFetch before rescore phase?");
         }
 
-        if (rescoreContext.getWindowSize() < topDocs.scoreDocs.length) {
-            throw new IllegalArgumentException(
-                "Rescore window is too small and should be at least the value of from + size but was ["
-                    + rescoreContext.getWindowSize()
-                    + "]"
-            );
-        }
-
         LocalModel definition = ltrRescoreContext.regressionModelDefinition;
 
-        // First take top slice of incoming docs, to be rescored:
-        TopDocs topNFirstPass = topN(topDocs, rescoreContext.getWindowSize());
```

Comment:
ℹ️ Since window size is always greater or equals than the topDocs, we can execute the rescoring on the full topDocs directly.